src/main/java/se/simonsoft/cms/transform/service/TransformServiceXsl.java

```diff
@@ -0,0 +1,289 @@
+import se.simonsoft.cms.item.info.CmsRepositoryLookup;
+import se.simonsoft.cms.item.properties.CmsItemPropertiesMap;
+import se.simonsoft.cms.transform.config.databind.TransformConfig;
+import se.simonsoft.cms.xmlsource.handler.XmlSourceReader;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceDocumentS9api;
+import se.simonsoft.cms.xmlsource.handler.s9api.XmlSourceReaderS9api;
+import se.simonsoft.cms.xmlsource.transform.SaxonOutputURIResolverXdm;
+import se.simonsoft.cms.xmlsource.transform.TransformOptions;
+import se.simonsoft.cms.xmlsource.transform.TransformStreamProvider;
+import se.simonsoft.cms.xmlsource.transform.TransformerService;
+import se.simonsoft.cms.xmlsource.transform.TransformerServiceFactory;
+
+public class TransformServiceXsl implements TransformService {
+	
+	private final CmsCommit commit;
+	private final CmsItemLookup itemLookup;
+	private final TransformerServiceFactory transformerServiceFactory;
+	private final CmsRepositoryLookup repoLookup;
+	private final Map<String, TransformerService> stylesheets;
+	private final XmlSourceReaderS9api sourceReader;
+	private final TransformerService transformerIdentity;
+	
+	private static final String TRANSFORM_LOCK_COMMENT = "Locked for transform";
+	private static final String TRANSFORM_BASE_PROP_KEY = "abx:TransformBase";
+	private static final String TRANSFORM_NAME_PROP_KEY = "abx:TransformName";
+	private static final Logger logger = LoggerFactory.getLogger(TransformServiceXsl.class);
+
+	@Inject
+	public TransformServiceXsl(
+			CmsCommit commit,
+			CmsItemLookup itemLookup,
+			CmsRepositoryLookup lookupRepo,
+			TransformerServiceFactory transfromerServiceFactory,
+			Map<String, TransformerService> stylesheets,
+			XmlSourceReader sourceReader
+			) {
+		
+		this.commit = commit;
+		this.itemLookup = itemLookup;
+		this.repoLookup = lookupRepo;
+		this.transformerServiceFactory = transfromerServiceFactory;
+		this.transformerIdentity = transfromerServiceFactory.buildTransformerService(new StreamSource(this.getClass().getClassLoader().getResourceAsStream("se/simonsoft/cms/xmlsource/transform/identity.xsl")));
+		this.stylesheets = stylesheets;
+		if (sourceReader instanceof XmlSourceReaderS9api) {
+			this.sourceReader = (XmlSourceReaderS9api) sourceReader;
+		} else {
+			throw new IllegalArgumentException("TransformServiceXsl requires Saxon S9API impl of XmlSourceReader, got: " + sourceReader.getClass());
+		}
+	}
+
+	@Override
+	public void transform(CmsItem item, TransformConfig config) {
+		if (config == null || config.getOptions() == null) {
+			throw new IllegalArgumentException("TransformServiceXsl needs a valid TransformConfig object.");
+		}
+		
+		final CmsItem base = item;
+		final CmsItemId baseItemId = base.getId();
+		final CmsRepository repository = baseItemId.getRepository();
+		final RepoRevision baseRevision = repoLookup.getYoungest(repository);
+		final Boolean overwrite = new Boolean(config.getOptions().getParams().get("overwrite"));
```

Comment:
Should this type be Boolean? Can it be null?

Reply:
Any value (Including null) added in Boolean constructor that is not equal to ignore case "true" is false.

Reply:
Use a boolean.